api/environments/models.py

```diff
@@ -358,6 +375,187 @@ def get_feature_state(
+
+        _features_aggregation_result: dict[str, int] = {}
+
+        # Closure to avoid recalculating twice the features metrics
+        def get_feature_agg(key: str) -> int:
+            nonlocal _features_aggregation_result
+            if not _features_aggregation_result:
+                _features_aggregation_result = features_qs.aggregate(
+                    total=models.Count("id"),
+                    enabled=models.Count("id", filter=Q(enabled=True)),
+                )
+            return _features_aggregation_result[key]
+
+        # Second optional callable is a function to override disabled - Initially to skip identity if not using edge
+        qs_map: dict[
+            EnvMetricsName, tuple[Callable[[], int], Callable[[], bool] | None]
+        ] = {
+            EnvMetricsName.TOTAL_FEATURES: (
+                lambda: get_feature_agg("total"),
+                None,
+            ),
+            EnvMetricsName.ENABLED_FEATURES: (
+                lambda: get_feature_agg("enabled"),
+                None,
+            ),
+            EnvMetricsName.SEGMENT_OVERRIDES: (
+                lambda: self._get_segment_metrics_queryset(with_workflows).count(),
+                None,
+            ),
+            EnvMetricsName.IDENTITY_OVERRIDES: (
+                (
+                    lambda: EdgeIdentity.dynamo_wrapper.get_identity_overrides_count_dynamo(
+                        self.api_key
+                    )
+                )
+                if self.project.enable_dynamo_db
+                else (lambda: self._get_identity_overrides_queryset().count()),
+                None,
+            ),
+        }
+
+        if with_workflows:
+            qs_map.update(
+                {
+                    EnvMetricsName.OPEN_CHANGE_REQUESTS: (
+                        lambda: self._get_change_requests_metrics_queryset().count(),
+                        None,
+                    ),
+                    EnvMetricsName.TOTAL_SCHEDULED_CHANGES: (
+                        lambda: self._get_scheduled_metrics_queryset().count(),
+                        None,
+                    ),
+                }
+            )
+
+        return build_metrics(qs_map)
+
+    def _get_identity_overrides_queryset(
+        self, with_workflows: bool = False
+    ) -> QuerySet[FeatureState]:
+        ids = self._get_active_feature_states_ids(
+            with_workflows,
+            "identity_id",
+            {"identity__isnull": False, "feature_segment__isnull": True},
+        )
+        result: QuerySet[FeatureState] = FeatureState.objects.filter(id__in=ids)
+        return result
+
+    def _get_active_feature_states_ids(
+        self,
+        with_workflows: bool = False,
+        extra_group_by_fields: Literal["identity_id"] | None = None,
+        filter_kwargs: dict[str, typing.Any] | None = None,
+    ) -> list[int]:
+        base_qs = FeatureState.objects.filter(
```

Comment:
You probably want to use `get_live_feature_states` method defined on `FeatureStateManager`, right?